### PR TITLE
CI: Replace Fedora 39 with 41

### PR DESCRIPTION
**Pull request by kinkie**
Fedora 39 will become EOL on 2024-11-12. Fedora 41 has been released.

### CODE REVIEW

**Comment by kinkie** [comment]
Workflow test run: https://github.com/kinkie/squid/actions/runs/11592207068

This will need to be backported to v6 after PR #1922 lands there

**Review by rousskov — APPROVED** [review]
Thank you for providing a link to staging test results! BTW, I suggest adding something like `cat /etc/issue` and/or `uname -a` to the beginning of Linux docker-based workflows to make it easier to validate that the right image was used. Right now, one would have to trust that the docker image was named correctly or fish for that virtualized OS information deep down in build logs (assuming it can be found at all). This suggestion/request is outside _this_ PR scope, of course.

In PR description, I removed a statement that duplicated PR title and provided a specific EOL date from Fedora 39 [schedule](https://fedorapeople.org/groups/schedule/f-39/f-39-key-tasks.html).

> This will need to be backported to v6

Your call. To me, it feels like v6 (that will be itself EOL soon) should be tested against its contemporary environment (to the extend that is feasible, of course), even if that environment becomes obsolete from current time point of view.

**Comment by kinkie** [comment]
> Thank you for providing a link to staging test results! BTW, I suggest adding something like `cat /etc/issue` and/or `uname -a` to the beginning of Linux docker-based workflows to make it easier to validate that the right image was used. Right now, one would have to trust that the docker image was named correctly or fish for that virtualized OS information deep down in build logs (assuming it can be found at all). This suggestion/request is outside _this_ PR scope, of course.

Sure, I can do that in a followup PR touching "test-builds.sh"; I would not do it as part of the test harness: your point is valid and it makes sense to do it always.

> In PR description, I fixed formatting, removed a statement that duplicated PR title, and provided a specific EOL date from Fedora 39 [schedule](https://fedorapeople.org/groups/schedule/f-39/f-39-key-tasks.html).

Thanks

> > This will need to be backported to v6
> 
> Your call. To me, it feels like v6 (that will be itself EOL soon) should be tested against its contemporary environment (to the extend that is feasible, of course), even if that environment becomes obsolete from current time point of view.

I see benefit in standardising as much as possible on a single solution

**Comment by rousskov** [comment]
> > Thank you for providing a link to staging test results! BTW, I suggest adding something like `cat /etc/issue` and/or `uname -a` to the beginning of Linux docker-based workflows to make it easier to validate that the right image was used. Right now, one would have to trust that the docker image was named correctly or fish for that virtualized OS information deep down in build logs (assuming it can be found at all). This suggestion/request is outside _this_ PR scope, of course.
> 
> Sure, I can do that in a followup PR touching "test-builds.sh"; I would not do it as part of the test harness: your point is valid and it makes sense to do it always.

I support adjusting test-builds.sh or even ./configure because that adjustment may be useful when folks share build logs from undisclosed or uncertain environments. Please be mindful of execution environments that lack /etc/issue. Thank you in advance for that followup PR!
 
> > > This will need to be backported to v6
> > 
> > Your call. To me, it feels like v6 (that will be itself EOL soon) should be tested against its contemporary environment (to the extend that is feasible, of course), even if that environment becomes obsolete from current time point of view.
> 
> I see benefit in standardising as much as possible on a single solution

This is not about an existence of some benefit. It is about balancing multiple benefits and harms. While it is always _possible_ to drop or add test platforms (or even stop testing!), I see more harm from not testing on a likely deployment target than benefit from using the same set of target for all Squid versions (i.e. the coveted "single solution"). For v6, a likely deployment targets probably include Fedora 39 (unless we are going to make an argument that we should not have tested on Fedora 39 to begin with).

**Comment by squidadm** [comment]
queued for backport to v6

**Pull request closed without merging**
